core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -588,9 +603,18 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
               if constexpr (ReduceFunctorHasFinal<FunctorType>::value)
                 FunctorFinal<FunctorType, WorkTag>::final(
                     static_cast<const FunctorType&>(functor), results_ptr);
+              if (device_accessible_result_ptr)
+                ValueOps::copy(functor, device_accessible_result_ptr,
```

Comment:
Why not directly `if (m_result_ptr_device_accessible) ValueOps::copy(functor, m_result_ptr ,`

Reply:
See https://github.com/kokkos/kokkos/pull/4089#discussion_r653582310.